x-pack/solutions/security/packages/kbn-cloud-security-posture/graph/src/components/graph_investigation/graph_investigation.tsx

```diff
@@ -138,11 +138,31 @@ export const GraphInvestigation = memo<GraphInvestigationProps>(
 
     const onInvestigateInTimelineCallback = useCallback(() => {
       const query = { ...kquery };
-      const filters = originEventIds.reduce<Filter[]>((acc, { id }) => {
-        return addFilter(dataView?.id ?? '', acc, EVENT_ID, id);
-      }, searchFilters);
 
-      if (query.query.trim() !== '' && originEventIds.length > 0) {
+      let filters = [...searchFilters];
+      // Case 1: Empty query with origin event IDs
+      if (query.query.trim() === '' && originEventIds.length > 0) {
```

Comment:
nit: you can simplify the if statement, `originEventIds.length > 0` is not relevant
the action fo the reduce in that case is similar

it is relevant to all the if statements here

if you wish to be explicit you can take it to another level instead of repeating it in all cases